### PR TITLE
bzl: add push rules for docker images

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: build/BUILD [hunk 1]
```diff
@@ -66,6 +77,20 @@ DOCKERIZED_BINARIES = {
     stamp = True,
 ) for binary in DOCKERIZED_BINARIES.keys()]
 
+[docker_bundle(
+    name = binary + "_dev",
+    images = {"gcr.io/mikedanese-k8s/%s:{STABLE_DOCKER_TAG}" % binary: binary + "-internal"},
```
Comment (mikedanese) [inline build/BUILD]: @ixdy @spxtr where should I put dev images?

File: build/BUILD [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
 
 load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build", "docker_bundle")
+load(
+    "@io_bazel_rules_docker//docker/contrib:push-all.bzl",
+    docker_pushall = "docker_push",
+)
 load("@io_kubernetes_build//defs:build.bzl", "release_filegroup")
 
+config_setting(
+    name = "actual_release",
```
Comment (mikedanese) [inline build/BUILD]: what should i call this setting?

**Comment by mikedanese** [comment]
@ixdy @spxtr ping. this is ready for review.

**Comment by spxtr** [comment]
Can you provide some usage information?

**Comment by mikedanese** [comment]
bazel run //build:push_all_images

File: build/BUILD [hunk 3]
```diff
@@ -160,3 +171,31 @@ filegroup(
         "//test/kubemark:all-srcs",
     ],
 )
+
+# These bundles group images at no build time cost unless a .tar is built from
+# the bundle. We use these bundles to push all images from the pushall rule.
+docker_bundle(
+    name = "all_images",
+    images = {
+        "gcr.io/google_containers/%s:{STABLE_DOCKER_TAG}" % (binary): binary + "-internal"
```
Comment (ixdy) [inline build/BUILD]: do we need to depend on the `-internal` one here? or would the result of `docker_bundle` work? I guess it probably doesn't matter?

File: build/BUILD [hunk 4]
```diff
@@ -160,3 +171,31 @@ filegroup(
...
+        for binary in DOCKERIZED_BINARIES.keys()
+    },
+    stamp = True,
+)
+
+docker_bundle(
+    name = "all_dev_images",
+    images = {
+        "gcr.io/kubernetes-ci-images/bazel/%s:{STABLE_DOCKER_TAG}" % (binary): binary + "-internal"
```
Comment (ixdy) [inline build/BUILD]: would it make more sense to use a "make variable" for the repo? I guess we don't have a good way of setting defaults, though.

**Comment by spxtr** [comment]
> bazel run //build:push_all_images

I mean, where do we document this sort of thing?

**Pull request closed without merging**